src/coreclr/pal/inc/pal_mstypes.h

```diff
@@ -543,8 +538,13 @@ typedef _W64 unsigned __int32 DWORD_PTR, *PDWORD_PTR;
 #define UlongToPtr(ul) ULongToPtr(ul)
 #define UintToPtr(ui)  UIntToPtr(ui)
 
-typedef ULONG_PTR SIZE_T, *PSIZE_T;
-typedef LONG_PTR SSIZE_T, *PSSIZE_T;
+#ifdef HOST_64BIT
+typedef unsigned long SIZE_T;
+typedef long SSIZE_T;
+#else
+typedef unsigned int SIZE_T;
+typedef int SSIZE_T;
+#endif
```

Comment:
This is very confusing at first glance. In order to make sure I get what is going on here, let me try to explain it.

This header is only compiled on non-Windows platforms, where `long` is 64-bit on a 64-bit architecture. This means we can define `SIZE_T` and `SSIZE_T` in terms of the platform's definition of `long`. If this code was run on the Windows platform this would be catastrophic.

If the above is indeed what I am seeing, I would ask for a comment to that effect. It might also be prudent to use a static assert in the spirit of

```cpp
static_assert(sizeof(SIZE_T) == sizeof(void*), "SIZE_T should be pointer sized");
```

Reply:
This actually doesn't change the definitions of `SIZE_T` and `SSIZE_T`. They were defined to the same thing before, only transitively. The `__int64` was defined as long on 64 bit Unix, and the `SIZE_T` was defined as `typedef ULONG_PTR SIZE_T` where `ULONG_PTR` was `typedef unsigned __int64 ULONG_PTR`. But now the `__int64` is changed to `long long` to match the OS definition and keeping `SIZE_T` defined the old way (thus becoming `unsigned long long`) has broken too many things.

Reply:
Nevertheless, adding such an assert is always a good idea.

Reply:
I've added the assert